apps/frontend-manage/src/components/activities/actions/useActivityLogAction.ts

```diff
@@ -0,0 +1,61 @@
+import { faMessage } from '@fortawesome/free-solid-svg-icons'
+import { ObjectType } from '@klicker-uzh/graphql/dist/ops'
+import { useTranslations } from 'next-intl'
+import { Dispatch, SetStateAction, useMemo } from 'react'
+import { ActivityAction } from './useAvailableActions'
+
+/**
+ * A hook that returns an activity log action for activity dropdown menus
+ * Can be used across all activity-related components to ensure consistency
+ */
+function useActivityLogAction({
+  objectId,
+  objectType,
+  setActivityLogOpen,
+}: {
+  objectId: string | number
+  objectType: ObjectType
+  setActivityLogOpen: Dispatch<SetStateAction<boolean>>
+}): ActivityAction {
+  const t = useTranslations()
+
+  return useMemo(() => {
+    // Get the display name for the object type
+    const objectTypeDisplay = (() => {
+      switch (objectType) {
```

Comment:
This does not seem ideal / necessary - the corresponding translations already exist and can be accessed using template strings
```
t(`shared.types.${objectType}`)
```